// containers/RegistrationContainer.jsx
import React, {Component} from "react";
import {connect} from "react-redux";
import {withRouter} from "react-router";
import {compose} from 'redux';

import Registration from 'components/registration';
import {registration, authorization} from 'store/actions';

class RegistrationContainer extends Component {

    state = {
        login: {
            username: '',
            password: ''
        },
        register: {
            username: '',
            password: ''
        }
    }

    submitForm = (event, form) => {
        event.preventDefault();
        const {username, password} = this.state[form];
        const {
            registration, 
            authorization, 
            history
        } = this.props;

        this.setState({
            [form]: {
                username: '',
                password: ''
            }
        });
        history.push('/');

        (form === 'register')
            ? registration(username, password)
            : authorization(username, password);
    }

    changeForm = (event, form) => {
        event.preventDefault();
        const {
            target: {
                name,
                value
            }
        } = event;
        this.setState({
            [form]: {
                ...this.state[form],
                [name]: value
            }
        });
    }

    render() {
        return <Registration
            submitForm={this.submitForm}
            changeForm={this.changeForm}
            state={this.state}/>
    }
}

function mapStateToProps({isAuth}) {
    return {isAuth};
}

function mapDispatchToProps(dispatch) {
    return {
        registration: (
            username, 
            password
            ) => dispatch(registration(
                username, 
                password
                )),
        authorization: (
            username, 
            password
            ) => dispatch(authorization(
                username, 
                password
                ))
    }
}

export default compose(
    withRouter, 
    connect(
        mapStateToProps, 
        mapDispatchToProps
    )
)(RegistrationContainer);
