// components/product-details/ProductDetails.jsx
import React from "react";
import StarRatings from 'react-star-ratings';
import PropTypes from 'prop-types';

import "./product-details.css";

export default function ProductDetails({
    product: {
        title,
        img,
        text
    },
    rating
}) {
    return (
        <section className="product-details">
            <div className="container mt-2">
                <div className="row align-items-center">
                    <div className="col-lg-5 col-md-5 col-sm-12 d-flex justify-content-center">
                        <img className="img-fluid" src={img} alt="product"/>
                    </div>
                    <div
                        className="col-lg-7 col-md-7 col-sm-12 text-lg-left text-sm-center text-md-left text-center ">
                        <h3 className="text-uppercase">{title}</h3>

                        <ul
                            className="rating-show navbar-nav flex-row justify-content-sm-center justify-content-md-start justify-content-center mt-3">
                            <StarRatings
                                rating={rating}
                                starRatedColor="#007bff"
                                starDimension="25px"
                                starSpacing="5px"/>
                        </ul>

                        <p className="pt-3">{text}</p>
                    </div>
                </div>
            </div>
        </section>
    )
}

ProductDetails.propTypes = {
    product: PropTypes.shape({
        title: PropTypes.string, 
        img: PropTypes.string, 
        text: PropTypes.string
    }),
    rating: PropTypes.number
}
