// components/comments-user/CommentsUser.jsx
import React from "react";
import StarRatings from "react-star-ratings";
import Moment from "react-moment";
import PropTypes from "prop-types";

import "./comments-user.css";

export default function CommentsUser({
  comment: {
    rate,
    text,
    created_at,
    created_by: { username }
  }
}) {
  return (
    <div className="container mt-4 mb-2">
      <div className="row align-items-center justify-content-center">
        <div className="col-lg-10 col-md-10 col-sm-12 m-2 pt-2 border rounded border-info">
          <div className="d-flex justify-content-between">
            <h5 className="text-capitalize text-info m-0">{username}</h5>
            <ul className="rating-show navbar-nav flex-row justify-content-sm-center justify-content-md-start">
              <StarRatings
                rating={rate}
                starRatedColor="#007bff"
                starDimension="20px"
                starSpacing="5px"
              />
            </ul>
          </div>
          <p className="text-secondary font-italic">
            <Moment format="DD.MM.YYYY, hh:mm">{created_at}</Moment>
          </p>
          <p>{text}</p>
        </div>
      </div>
    </div>
  );
}

CommentsUser.propTypes = {
  comment: PropTypes.shape({
    rate: PropTypes.number,
    text: PropTypes.string,
    created_at: PropTypes.string,
    created_by: PropTypes.shape({ username: PropTypes.string })
  })
};
