// components/product-list-item/ProductListItem.jsx
import React, {Component} from "react";
import {withRouter} from "react-router";
import PropTypes from 'prop-types';

import "./product-list-item.css";

class ProductListItem extends Component {
    static propTypes = {
        history: PropTypes.shape({
            push: PropTypes.func,
        }),
        product: PropTypes.shape({
            id: PropTypes.number, 
            title: PropTypes.string, 
            text: PropTypes.string, 
            img: PropTypes.string
        })
    }

    goProductPage = () => {
        const {product: {
                id
            }, history} = this.props;
            
        history.push(`/product/${id}`)
    }
    render() {
        const {title, text, img} = this.props.product;
        return (
            <section className="product">
                <div className="container mt-2">
                    <div className="row align-items-center justify-content-center">
                        <div>
                            <img className="img-fluid" src={img} alt="product"/>
                        </div>
                        <div
                            className="col-lg-7 col-md-7 col-sm-12 text-lg-left text-sm-center text-md-left text-center">
                            <h3 className="text-uppercase pt-4">{title}</h3>
                            <p className="pt-2">{`${text.substr(0, 60)}...`}</p>
                            <div>
                                <button
                                    className="btn btn-outline-info btn-lg mr-3"
                                    onClick={this.goProductPage}>See details</button>
                            </div>
                        </div>
                    </div>
                </div>
            </section>
        );
    }

}

export default withRouter(ProductListItem);
