// components/registration/Registration.jsx
import React from 'react';
import PropTypes from 'prop-types';

import RegisterItem from 'components/register-item'

import './registration.css'

export default function Registration({
    changeForm,
    submitForm,
    state: {
        login,
        register
    }
}) {

    return (
        <section className="sing bg-light m-5">
            <div className="container mt-4 mb-2">
                <div className="row align-items-center justify-content-around">
                    <div className="col-lg-5 col-md-5 col-sm-12 mt-2">
                        <RegisterItem
                            titleForm="LOGIN"
                            titleBtn="GO"
                            handleChange={event => changeForm(event, 'login')}
                            handleSubmit={event => submitForm(event, 'login')}
                            value={login}/>
                    </div>
                    <div
                        className="col-lg-2 col-md-2 col-sm-12 mt-2 d-flex justify-content-center align-items-center">
                        <div className="or border border-info rounded-circle text-center">OR</div>
                    </div>
                    <div className="col-lg-5 col-md-5 col-sm-12 mt-2">
                        <RegisterItem
                            titleForm="REGISTER"
                            titleBtn="SING UP"
                            handleChange={event => changeForm(event, 'register')}
                            handleSubmit={event => submitForm(event, 'register')}
                            value={register}/>
                    </div>
                </div>
            </div>
        </section>
    )
}

Registration.propTypes = {
    changeForm: PropTypes.func,
    submitForm: PropTypes.func,
    state: PropTypes.shape({
        login: PropTypes.shape({
            login: PropTypes.string,
            register: PropTypes.string,
        }),
        register: PropTypes.shape({
            login: PropTypes.string,
            register: PropTypes.string,
        }),
    }),
}
