// store/actions/actionCreators.js
import {
    FETCH_PRODUCTS_REQUEST,
    FETCH_PRODUCTS_SUCCESS,
    FETCH_PRODUCTS_FAILURE,

    FETCH_PRODUCT_DETAILS_REQUEST,
    FETCH_PRODUCT_DETAILS_SUCCESS,
    FETCH_PRODUCT_DETAILS_FAILURE,

    FETCH_COMMENTS_REQUEST,
    FETCH_COMMENTS_SUCCESS,
    FETCH_COMMENTS_FAILURE,

    REGISTRATION_REQUEST,
    REGISTRATION_SUCCESS,
    REGISTRATION_FAILURE,

    LOGOUT,

    AUTHORIZATION_REQUEST,
    AUTHORIZATION_SUCCESS,
    AUTHORIZATION_FAILURE,

    SEND_FEEDBACK_REQUEST,
    SEND_FEEDBACK_SUCCESS,
    SEND_FEEDBACK_FAILURE

} from 'store/actions/actionTypes';

// FETCH_PRODUCTS

export const productsRequested = () => {
    return {type: FETCH_PRODUCTS_REQUEST};
};
export const productsLoaded = (data) => {
    return {
        type: FETCH_PRODUCTS_SUCCESS,
        payload: {
            data: data
        }
    };
};
export const productsError = (error) => {
    return {
        type: FETCH_PRODUCTS_FAILURE,
        payload: {
            data: error
        }
    };
};

// FETCH_PRODUCT_DETAILS

export const productDetailsRequested = () => {
    return {type: FETCH_PRODUCT_DETAILS_REQUEST};
};
export const productDetailsLoaded = (data) => {
    return {
        type: FETCH_PRODUCT_DETAILS_SUCCESS,
        payload: {
            data: data
        }
    };
};
export const productDetailsError = (error) => {
    return {
        type: FETCH_PRODUCT_DETAILS_FAILURE,
        payload: {
            data: error
        }
    };
};

// FETCH_COMMENTS

export const coommentsRequested = () => {
    return {type: FETCH_COMMENTS_REQUEST};
};
export const coommentsLoaded = (data) => {
    return {
        type: FETCH_COMMENTS_SUCCESS,
        payload: {
            data: data
        }
    };
};
export const coommentsError = (error) => {
    return {
        type: FETCH_COMMENTS_FAILURE,
        payload: {
            data: error
        }
    };
};

// REGISTRATION

export const registrationRequested = () => {
    return {type: REGISTRATION_REQUEST};
};
export const registrationLoaded = (data, username) => {
    return {
        type: REGISTRATION_SUCCESS,
        payload: {
            token: data.token,
            success: data.success,
            username
        }
    };
};
export const registrationError = (error) => {
    return {
        type: REGISTRATION_FAILURE,
        payload: {
            data: error
        }
    };
};

// LOGOUT

export const logoutCreate = () => {
    return {type: LOGOUT};
};

// AUTHORIZATION

export const authorizationRequested = () => {
    return {type: AUTHORIZATION_REQUEST};
};
export const authorizationLoaded = (data, username) => {
    return {
        type: AUTHORIZATION_SUCCESS,
        payload: {
            token: data.token,
            success: data.success,
            username
        }
    };
};
export const authorizationError = (error) => {
    return {
        type: AUTHORIZATION_FAILURE,
        payload: {
            data: error
        }
    };
};

// SENDFEEDBACK

export const sendFeedbackRequested = () => {
    return {type: SEND_FEEDBACK_REQUEST};
};
export const sendFeedbackLoaded = (data) => {
    return {
        type: SEND_FEEDBACK_SUCCESS,
        payload: {
            data: data
        }
    };
};
export const sendFeedbackError = (error) => {
    return {
        type: SEND_FEEDBACK_FAILURE,
        payload: {
            data: error
        }
    };
};
