// components/register-item/RegisterItem.jsx
import React from "react";
import PropTypes from "prop-types";

import "./register-item.css";

export default function RegisterItem({
  titleForm,
  titleBtn,
  handleChange,
  handleSubmit,
  value: { username, password }
}) {
  return (
    <div className="jumbotron mb-2">
      <h5 className="text-capitalize text-info mb-5 text-center">
        {titleForm}
      </h5>
      <form
        autoComplete="off"
        className="form accountbox"
        onSubmit={handleSubmit}
      >
        <div className="">
          <input
            className="pl-3 mt-2 border rounded border-info"
            type="text"
            name="username"
            onChange={handleChange}
            value={username}
            placeholder="* User Name"
            required
          />
        </div>
        <div>
          <input
            className="input-block pl-3 mt-4 border rounded border-info"
            type="password"
            name="password"
            onChange={handleChange}
            value={password}
            placeholder="* Password"
            required
          />
        </div>
        <div>
          <button
            type="submit"
            className="btn btn-outline-success btn-block mt-4 mb-4"
          >
            {titleBtn}
          </button>
        </div>
      </form>
    </div>
  );
}

RegisterItem.propTypes = {
  titleForm: PropTypes.string,
  titleBtn: PropTypes.string,
  handleChange: PropTypes.func,
  handleSubmit: PropTypes.func,
  value: PropTypes.shape({
    username: PropTypes.string,
    password: PropTypes.string
  })
};
