// services/fetchDataApi.js
import axios from 'axios';

import {getMockData} from 'services/mockyData';

export const fetchProductsApi = async() => {
    const {data} = await axios.get("/api/products/");
    getMockData(data);
    return data;
}

export const fetchProductDetailsApi = async(id) => {
    const {data} = await axios.get("/api/products/");
    getMockData(data);
    const newData = data.find((item) => item.id === + id);
    return newData;
}

export const fetchComentsApi = async(id) => {
    const {data} = await axios.get(`/api/reviews/${id}`);
    return data;
}

export const registrationApi = async(username, password) => {
    const {data} = await axios(`/api/register/`, {
        method: 'POST',
        headers: {
            'Accept': 'application/json',
            'Content-Type': "application/json"
        },
        data: JSON.stringify({
            "username": username, 
            "password": password
        })
    });
    return data;
}

export const authorizationApi = async(username, password) => {
    const {data} = await axios(`/api/login/`, {
        method: 'POST',
        headers: {
            'Accept': 'application/json',
            'Content-Type': "application/json"
        },
        data: JSON.stringify({
            "username": username, 
            "password": password
        })
    });
    return data;
}

export const sendFeedbackApi = async(id, rate, text) => {
    const {data} = await axios(`/api/reviews/${id}`, {
        method: 'POST',
        headers: {
            'Content-Type': "application/json"
        },
        data: JSON.stringify({
            "rate": rate, 
            "text": text
        })
    });
    return data;
}
