// components/comment-for-posting/CommentForPosting.jsx
import React from "react";
import StarRatings from 'react-star-ratings'
import PropTypes from 'prop-types';

import './comment-for-posting.css'

export default function CommentForPosting({
    setRating, 
    setComment, 
    handleChangeRating, 
    handleCommentChange, 
    handleSubmit
}) {
    return (
        <form className="form-comment" onSubmit={handleSubmit}>
            <div className="row align-items-center justify-content-center">
                <div className="col-lg-7 col-md-7 col-sm-12 mt-2">
                    <div className="form-group">
                        <textarea
                            required
                            name="rating"
                            className="form-control form-control-lg"
                            placeholder="Your Comment"
                            rows="4"
                            onChange={handleCommentChange}
                            value={setComment}></textarea>
                    </div>
                </div>
                <div
                    className="col-lg-3 col-md-3 col-sm-12 d-flex flex-column align-items-center">
                    <div
                        className="d-flex justify-content-sm-center justify-content-md-start justify-content-center ml-2 mb-2">
                        <StarRatings
                            rating={setRating}
                            starRatedColor="#007bff"
                            starHoverColor="#28a745"
                            starDimension="22px"
                            starSpacing="4px"
                            changeRating={handleChangeRating}
                            name="rating"/>
                    </div>
                    <div
                        className="d-flex justify-content-sm-center justify-content-md-start justify-content-center">
                        <input
                            type="submit"
                            className="btn btn-outline-info btn-lg mt-2"
                            value="Add Comment"
                            name="rating"/>
                    </div>
                </div>
            </div>
        </form>
    )
}

CommentForPosting.propTypes = {
    setRating: PropTypes.number,
    setComment: PropTypes.string,
    handleChangeRating: PropTypes.func,
    handleCommentChange: PropTypes.func,
    handleSubmit: PropTypes.func
}
